packages/hoppscotch-common/src/helpers/import-export/export/postman.ts

```diff
@@ -0,0 +1,184 @@
+  if (hoppBody.contentType === null) {
+    return undefined
+  } else if (hoppBody.contentType === "multipart/form-data") {
+    return {
+      mode: "formdata",
+      formdata: hoppBody.body.map((kv) => ({
+        key: hoppToPostmanTemplating(kv.key),
+        // kv.value can also be Blob[] and we aren't storing it
+        value:
+          typeof kv.value === "string" ? hoppToPostmanTemplating(kv.value) : "",
+      })),
+    }
+  } else if (hoppBody.contentType === "application/x-www-form-urlencoded") {
+    const maybeBody = parseRawKeyValueEntriesE(hoppBody.body)
+    if (isLeft(maybeBody)) {
+      return undefined
+    }
+    return {
+      mode: "urlencoded",
+      urlencoded: maybeBody.right.map((kv) => ({
+        key: hoppToPostmanTemplating(kv.key),
+        value: hoppToPostmanTemplating(kv.value),
+      })),
+    }
+  } else {
+    return { mode: "raw", raw: hoppBody.body }
+  }
+}
+
+const getHeaders = (
+  request: HoppRESTRequest
+): { key: string; value: string }[] => {
+  let out = []
+  for (const header of request.headers) {
+    out.push({ key: header.key, value: hoppToPostmanTemplating(header.value) })
+  }
+
+  if (
+    request.body.contentType !== null &&
+    request.body.contentType !== "multipart/form-data"
+  ) {
+    // Postman took the first value, hence removing
+    out = out.filter((header) => header.key.toLowerCase() !== "content-type")
+    out.push({ key: "Content-Type", value: request.body.contentType })
+  }
+
+  return out
+}
+
+function parseMyCollection(
+  hoppCollection: HoppCollection<HoppRESTRequest>
+): ItemGroup<Item> {
+  const postmanItemGoup = new ItemGroup<Item>()
+  postmanItemGoup.name = hoppCollection.name
+  for (const folder of hoppCollection.folders) {
+    postmanItemGoup.items.add(parseMyCollection(folder))
+  }
+  for (const request of hoppCollection.requests) {
+    postmanItemGoup.items.add(hoppRequestToPostmanRequest(request))
+  }
+
+  return postmanItemGoup
+}
+
+function hoppRequestToPostmanRequest(request: HoppRESTRequest): Item {
+  const url = new URL(request.endpoint)
+  const postmanRequest = new Item({
+    request: {
+      url: {
+        hash: url.hash,
+        host: url.host,
+        path: hoppToPostmanTemplating(url.pathname),
+        // Postman should not get a "" value as it will create url of host:/path instead of host/path
+        port: url.port === "" ? undefined : url.port,
+        // Todo we are not caring about the active status
+        query: request.params.map((param) => ({
+          key: hoppToPostmanTemplating(param.key),
+          value: hoppToPostmanTemplating(param.value),
+        })),
+        protocol: url.protocol.replace(":", ""), // Got a value like https:
+      },
+      method: request.method,
+      // Todo we are not caring about the active status of header
+      header: getHeaders(request),
+      body: getBody(request.body),
+      auth: getAuth(request),
+    },
+  })
+  postmanRequest.name = request.name
```

Comment:
Why not set the name in the constructor itself ?

```suggestion
  const postmanRequest = new Item({
    name: request.name,
    request: {
      url: {
        hash: url.hash,
        host: url.host,
        path: hoppToPostmanTemplating(url.pathname),
        // Postman should not get a "" value as it will create url of host:/path instead of host/path
        port: url.port === "" ? undefined : url.port,
        // Todo we are not caring about the active status
        query: request.params.map((param) => ({
          key: hoppToPostmanTemplating(param.key),
          value: hoppToPostmanTemplating(param.value),
        })),
        protocol: url.protocol.replace(":", ""), // Got a value like https:
      },
      method: request.method,
      // Todo we are not caring about the active status of header
      header: getHeaders(request),
      body: getBody(request.body),
      auth: getAuth(request),
    },
  })
```

Reply:
```
export interface ItemDefinition extends PropertyDefinition {
  request?: RequestDefinition | undefined;
  response?: ResponseDefinition[] | undefined;
  events?: EventDefinition[] | undefined;
}

export class Item extends Property<ItemDefinition> {
  events: EventList;
  request: Request;
  responses: PropertyList<Response>;

  constructor(definition?: ItemDefinition);

  authorizeRequestUsing(type: string | RequestAuthDefinition, options?: VariableList): void;

  getAuth(): RequestAuth;

  getEvents(name?: string): Event[];

  static isItem(obj: any): boolean;
}
```

Seems name is not in any constructor.

Reply:
If you went for that approach to justify, you left out the actually important bit. 🤦 

<img width="1001" alt="image" src="https://user-images.githubusercontent.com/9131943/221833816-9debedeb-8bdc-47c6-8df1-1b3981a3d793.png">